core/station/impl/src/services/system.rs

```diff
@@ -386,25 +446,71 @@ impl SystemService {
     pub async fn init_canister(&self, input: SystemInit) -> ServiceResult<()> {
         let mut system_info = SystemInfo::default();
 
-        if input.admins.is_empty() {
-            return Err(SystemError::NoAdminsSpecified)?;
-        }
-
-        if input.admins.len() > u16::MAX as usize {
-            return Err(SystemError::TooManyAdminsSpecified {
-                max: u16::MAX as usize,
-            })?;
+        match &input.initial_config {
+            InitialConfig::WithAllDefaults {
+                admin_quorum,
+                operator_quorum,
+                users,
+            } => {
+                // adds the default admin group
+                init_canister_sync_handlers::add_default_groups();
+                // registers the admins of the canister
+                init_canister_sync_handlers::set_initial_users(users.clone(), &[*ADMIN_GROUP_ID])?;
+                // registers the default canister configurations such as policies and user groups.
+                init_canister_sync_handlers::init_default_permissions_and_policies(
+                    *admin_quorum,
+                    *operator_quorum,
+                )?;
+                // add default assets
+                init_canister_sync_handlers::add_default_assets();
+            }
+            InitialConfig::WithDefaultPolicies {
+                assets,
+                users,
+                admin_quorum,
+                operator_quorum,
+                ..
+            } => {
+                // adds the default admin group
+                init_canister_sync_handlers::add_default_groups();
+                // registers the admins of the canister
+                init_canister_sync_handlers::set_initial_users(users.clone(), &[*ADMIN_GROUP_ID])?;
```

Comment:
should we fail here if the users have the `groups` filled up and they are not ADMIN_GROUP_ID or OPERATOR_GROUP_ID? i am wondering this because other then those two, with the `WithDefaultPolicies` and `WithAllDefaults` variants there is no way to provide more user groups